templates/fragments.txt

```diff
@@ -0,0 +1,137 @@
+{% macro uart_define() %}
+// UART defines
+// By default the stdout UART is `uart0`, so we will use the second one
+#define UART_ID uart1
+#define BAUD_RATE 9600
+// Use pins 4 and 5 for UART1
+// Pins can be changed, see the GPIO function select table in the datasheet for information on GPIO assignments
+#define UART_TX_PIN 4
+#define UART_RX_PIN 5 
+{% endmacro %}
+
+{% macro uart_initialiser() %}
+// Set up our UART
+uart_init(UART_ID, BAUD_RATE);
+// Set the TX and RX pins by using the function select on the GPIO
+// Set datasheet for more information on function select
+gpio_set_function(UART_TX_PIN, GPIO_FUNC_UART);
+gpio_set_function(UART_RX_PIN, GPIO_FUNC_UART);
+{% endmacro %}
+
+{% macro spi_define() %}
+// SPI Defines
+// We are going to use SPI 0, and allocate it to the following GPIO pins,
+// Pins can be changed, see the GPIO function select table in the datasheet for information on GPIO assignments,
+#define SPI_PORT spi0
+#define PIN_MISO 16
+#define PIN_CS   17
+#define PIN_SCK  18
+#define PIN_MOSI 19 
+{% endmacro %}
+
+{% macro spi_initialiser() %}
+// SPI initialisation. This example will use SPI at 1MHz.
+spi_init(SPI_PORT, 1000*1000);
+gpio_set_function(PIN_MISO, GPIO_FUNC_SPI);
+gpio_set_function(PIN_CS,   GPIO_FUNC_SIO);
+gpio_set_function(PIN_SCK,  GPIO_FUNC_SPI);
+gpio_set_function(PIN_MOSI, GPIO_FUNC_SPI); 
+// Chip select is active-low, so we'll initialise it to a driven-high state
+gpio_set_dir(PIN_CS, GPIO_OUT);
+gpio_put(PIN_CS, 1);
+{% endmacro %}
+
+{% macro i2c_define() %}
+// I2C defines
+// This example will use I2C0 on GPIO8 (SDA) and GPIO9 (SCL) running at 400KHz.
+// Pins can be changed, see the GPIO function select table in the datasheet for information on GPIO assignments
+#define I2C_PORT i2c0
+#define I2C_SDA 8
```

Comment:
@kilograham I think this project-generator was created before we added the "board headers" to the Raspberry Pi Pico SDK. Should we now update the pin-numbers used here (for the various interfaces) to match up with https://github.com/raspberrypi/pico-sdk/blob/master/src/boards/include/boards/pico.h ?

Reply:
Needs to be a separate issue.